File: Part1_강의자료/Ch08_투포인터/문제별코드/14465_소가길을건너간이유5/14465_sliding_window.py
N, K, B = map(int, input().split())
is_broken = [0] * (N + 1)
for _ in range(B):
    is_broken[int(input())] = 1

broken_cnt = sum(is_broken[1:K+1])
ans = broken_cnt
for i in range(2, N - K + 2):
    broken_cnt += is_broken[i + K - 1] - is_broken[i - 1]
    ans = min(ans, broken_cnt)

print(ans)

File: Part1_강의자료/Ch08_투포인터/문제별코드/14465_소가길을건너간이유5/14465_sliding_window.cpp
#include <algorithm>
#include <stdio.h>
#include <vector>
using namespace std;

int main()
{
    int N, K, B;
    scanf("%d %d %d", &N, &K, &B);

    int b;
    vector<bool> is_broken(N + 1);
    while (B--)
    {
        scanf("%d", &b);
        is_broken[b] = true;
    }

    int broken_cnt = 0;
    for (int i = 1; i <= K; i++)
        broken_cnt += is_broken[i];

    int ans = broken_cnt;
    for (int i = 2; i <= N - K + 1; i++)
    {
        broken_cnt += is_broken[i + K - 1] - is_broken[i - 1];
        ans = min(ans, broken_cnt);
    }
    printf("%d\n", ans);
    return 0;
}
